pkg/customresourcestate/config.go

```diff
@@ -159,13 +160,27 @@ type ConfigDecoder interface {
 }
 
 // FromConfig decodes a configuration source into a slice of customresource.RegistryFactory that are ready to use.
-func FromConfig(decoder ConfigDecoder) ([]customresource.RegistryFactory, error) {
+func FromConfig(decoder ConfigDecoder, r *discovery.GVKMaps) ([]customresource.RegistryFactory, error) {
 	var crconfig Metrics
 	var factories []customresource.RegistryFactory
 	factoriesIndex := map[string]bool{}
 	if err := decoder.Decode(&crconfig); err != nil {
 		return nil, fmt.Errorf("failed to parse Custom Resource State metrics: %w", err)
 	}
+	var resources []Resource
+	for _, resource := range crconfig.Spec.Resources {
+		gotGVKs := r.ResolveGVK(resource.GroupVersionKind)
+		if len(gotGVKs) >= 2 {
```

Comment:
Can you add a comment here why `>= 2` ?

Reply:
I refactored the logic a bit further, PLMK if a comment is still needed and I'll add it.